test/test_regression_cli.py

```diff
@@ -1,153 +1,136 @@
-        os.system("rm output_file*")
-        os.system('mysql -u travis -Bse "DROP DATABASE IF EXISTS testdb"') # installing over an existing database changes the dump
-        os.system("retriever install mysql %s -u travis -d testdb" %dataset)  # user 'travis' for Travis CI
-        os.system("mysqldump testdb -u travis --compact --compatible=no_table_options --no-create-db --no-create-info --result-file=output_file")
-        current_md5 = getmd5(unixfileformat("output_file"))
-        assert current_md5 == known_md5
-
-
-class PostgreSQLRegression(TestCase):
-
-    def check_postgres_regression(self, dataset, known_md5):
-        """Check for regression for a particular dataset imported to postgres"""
-        os.system("rm output_file*")
-        os.system('psql -U postgres -d testdb -h localhost -c "DROP SCHEMA IF EXISTS testschema CASCADE"')
-        os.system("retriever install postgres %s -u postgres -d testdb -a testschema" % dataset)
-        os.system("pg_dump -n testschema --data-only -U postgres -h localhost -f output_file testdb")
-        current_md5 = getmd5(unixfileformat("output_file"))
-        assert current_md5 == known_md5
-
-
-class DownloadRegression(TestCase):
-
-    def check_download_regression(self, dataset, known_md5):
-        """Check for regression for a particular dataset downloaded only"""
-        os.system("retriever download {0} -p raw_data/{0}".format(dataset))
-        current_md5 = getmd5("raw_data/%s" % (dataset))
-        assert current_md5 == known_md5
-
-
-dbms_test_classes = {'sqlite': SqliteRegression, 'csv': CSVRegression,
-                     'mysql': MySQLRegression, 'postgres': PostgreSQLRegression,
-                     'download': DownloadRegression}
-
-for dbms in known_md5s:
-    for dataset in known_md5s[dbms]:
-        stub_test = _test_factory('check_%s_regression' % dbms, 'test_%s' % dataset,
-                                  dataset, known_md5s[dbms][dataset])
-        setattr(dbms_test_classes[dbms], stub_test.__name__, stub_test)
-        del(stub_test)
-
-if __name__ == '__main__':
-    nose.runmodule()
+def tocsv(arg=None, mode="rb"):
+    # at this point the current directory is ./test
+    if os.path.exists("output_dumps"):
+        os.system("rm -r output_dumps")
+    os.makedirs("output_dumps")
+    os.chdir("output_dumps")
+    if arg:
+        engine = choose_engine(arg.__dict__)
+        script_list = SCRIPT_LIST()
+        scripts = name_matches(script_list, arg.dataset)
+        for dataset_i in scripts:
+            if arg.engine in ["csv", 'json', 'xml']:
+                # the engines in the list need to download the data
+                dataset_i.download(engine, arg.debug, False)
+                dataset_i.engine.final_cleanup()
+
+            dataset_i.download(engine, arg.debug, True)
+            dataset_i.engine.final_cleanup()
+    os.chdir("..")
+    current_md5 = getmd5("output_dumps", mode)
+    return current_md5
```

Comment:
I'm confused as to why we're handling csv, json, and xml differently than the other engines. I would have installed them using the CLI in the tests below just like you did for the database engines. Can you explain why we're installing them from here instead?